main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	// "db"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+)
+
+func main() {
+	router := gin.Default()
+	fmt.Println("Hello World!")
+
+	router.Use(WelcomeEndpoint)
+
+	router.GET("/v1/welcome", func(c *gin.Context) {
+		if c.FullPath() == "/v1/welcome" {
```

Comment:
Here you are checking the API path in API itself. So it always is true.
You have to create one middleware that will be used by both APIs `welcome` and `hello world`, and in that middleware, you have to check which API has been called.